highfidelity/hifi
# Change the way cellscience entities intitialize & audio plays

Changelist:
- Previously, entities in cell science tried to initialize themselves using userData in their preload methods.  Since userData was sometimes inaccessible at preload, those entities would fail to do their assigned behaviors.  Now these entities wait until they are able to access userData to initialize themselves, which should help a variety of issues. 
- Previously, over 125mb of  background audio was downloaded and loaded locally by each user because audio was played through entities.  Now audio is injected on the AC, with no user download required.
- Previously, some entities had movement scripts attached where everyone tried to move the entity in its update, causing jitter.  Now only one person at a time can move any given entity in its update.
- Halved the number of vesicles and other entities for performance reasons.
- Various cleanup and minor changes, including making the entities at the cell level 'zoomable' when you enter them.

**NOTE: This script will clear content where you run it**

To test: 

1. In an empty domain, go to running scripts and import this script from url: http://hifi-content.s3.amazonaws.com/DomainContent/CellScience/importCellScience.js
2. Stop the importCellScience.js script.
3. Observe that all the buttons work, background audio is playing at a talkable volume, and that from the 'Cells' button area when you fly into an entity, you get teleported deeper into the cell,.
